### DISCUSSION TITLE
calcGrain fills empty data points

### DISCUSSION THREAD

**Discussion by huseyinhzl** [mtex-toolbox/mtex#1954] [Ask Anything]
Hello all,

I have a question about calcGrains. I am computing grains with the following command, but after computation, it fills the voids in the EBSD map with the grains. I have attached the corresponding maps. I am wondering how I can get rid of the filling voids in the EBSD maps.

Thank you

```
% compute the grains
% Grains are calculated with a 1.5° threshold
[grains,ebsdP.grainId] = calcGrains(ebsdP('indexed'),'angle',1.5*degree,'boundary','tight',...
     'removeQuadruplePoints');
% EBSD data in small grains are removed
ebsdP(grains(grains.grainSize < 3)) = [];
% Recalculate the grains from the remaining data ...
[grains,ebsdP.grainId] = calcGrains(ebsdP('indexed'),'angle',1.5*degree,'boundary','tight',...
     'removeQuadruplePoints');
% ... and smooth the grain boundaries
grains = smooth(grains,5);
```

![Raw EBSD](https://github.com/mtex-toolbox/mtex/assets/120036622/5b8d9fd7-269f-4a3b-bfd4-5d67c1f5dd58)
![Grains](https://github.com/mtex-toolbox/mtex/assets/120036622/27cec175-d6b9-407d-b9f3-8b23656b6a98)

**Comment by kilir** [comment]
Hi,
if your points exists as 'nonIndexed', you should be fine with 
```
[grains,ebsdP.grainId] = calcGrains(ebsdP,'angle',1.5*degree,'boundary','tight',...
     'removeQuadruplePoints');
```

otherwise you could gridify your data and set the nan-rotations to the nonIndexed phase or you can use the option 'unitCell' if you want to stick to the mapping grid.

Cheers,
Rüdiger
